storage/store/latest_persisted_sealed_result_test.go

```diff
@@ -111,38 +113,46 @@ func TestLatestPersistedSealedResult_BatchSet(t *testing.T) {
 		db := pebbleimpl.ToDB(pdb)
 
 		t.Run("successful batch update", func(t *testing.T) {
+			lockManager := storage.NewTestingLockManager()
+
 			initializer := NewConsumerProgress(db, "test_consumer1")
 			progress, err := initializer.Initialize(initialHeader.Height)
 			require.NoError(t, err)
 
 			latest, err := NewLatestPersistedSealedResult(progress, mockHeaders, mockResults)
 			require.NoError(t, err)
 
-			batch := db.NewBatch()
-			defer batch.Close()
+			err = unittest.WithLock(t, lockManager, storage.LockInsertCollection, func(lctx lockctx.Context) error {
+				batch := db.NewBatch()
+				defer batch.Close()
```

Comment:
nit: we could replace the `NewBatch`, `Close` and `Commit` with `return db.WithReaderBatchWriter(batch storage.ReaderBatchWriter) error {`